src/Components/Catalog/Carousel.tsx

```diff
@@ -0,0 +1,349 @@
+import { useState, useEffect, useMemo } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { FaArrowLeft, FaArrowRight } from "react-icons/fa";
+import {
+  selectCarouselFilter,
+  selectCarouselFilterType,
+} from "../../redux/carousel/carouselSelectors";
+import { setFilter, setFilterType } from "../../redux/carousel/carouselSlice";
+import ProductData from "./Carousel.json";
+
+// Helper function to normalize strings for comparison
+const normalizeString = (str: string) => {
+  if (!str) return "";
+  let normalized = str.toLowerCase().trim(); // Remove 's' at the end if present
+  normalized = normalized.replace(/s$/, ""); // Handle common plural forms
+  normalized = normalized.replace(/\s+/g, ""); // Remove spaces
+  return normalized;
+};
+
+const findBestMatch = (filter: string, filterType: string) => {
+  const normalizedFilter = normalizeString(filter);
+
+  return ProductData.filter((item) => {
+    if (filterType === "category") {
+      return (
+        normalizeString(item.category).includes(normalizedFilter) ||
+        normalizedFilter.includes(normalizeString(item.category))
+      );
+    } else if (filterType === "brand") {
+      return (
+        normalizeString(item.brand).includes(normalizedFilter) ||
+        normalizedFilter.includes(normalizeString(item.brand))
+      );
+    }
+    return false;
+  });
+};
```

Comment:
i think this extra bit of processing can be fully avoided if we just stick to following static hardcoded names for categories and brands, so no need to find best match, simply use the category/brand key to filter product data